### PR TITLE
Task: Debugging Init Containers

### PR DESCRIPTION
**Pull request by enisoc**
cc @erictune @foxish @kow3ns @janetkuo @kubernetes/sig-apps

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1821)

### CODE REVIEW

**Comment by enisoc** [comment]
Here's a direct link to the preview: https://deploy-preview-1821--kubernetes-io-vnext-staging.netlify.com/docs/tasks/troubleshoot/debug-init-containers/

File: docs/tasks/troubleshoot/debug-init-containers.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+A Pod with Init Containers may show the following special status values:
+
+Status | Meaning
+------ | -------
+`Init:N/M` | The Pod has `M` Init Containers, and `N` have completed so far.
+`Init:Error` | An Init Container has failed to execute.
+`Init:CrashLoopBackOff` | An Init Container has failed repeatedly.
+
+A Pod with status `Pending` has not yet begun executing Init Containers.
```
Comment (foxish) [inline docs/tasks/troubleshoot/debug-init-containers.md]: We could instead link to http://kubernetes.io/docs/user-guide/pod-states/

Comment (enisoc) [inline docs/tasks/troubleshoot/debug-init-containers.md]: Are you saying we should add this information to that page and then link to it? As it is, that page doesn't say anything about Init Containers.

Comment (foxish) [inline docs/tasks/troubleshoot/debug-init-containers.md]: On second thought, it probably makes sense here. However, I think init-containers can show [other errors](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/images/types.go#L27-L42) similar to pod containers. The list above is not exhaustive. We should maybe change the wording to say that those are some of the common error states. I can't seem to find anywhere with all the possible states we could be in.

**Comment by foxish** [comment]
LGTM

File: docs/tasks/troubleshoot/debug-init-containers.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+Init Containers.
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+* You should be familiar with the basics of
+  [Init Containers](/docs/user-guide/pods/init-containers/).
+* You should have a [Pod](/docs/user-guide/pods/) you want to debug that uses
+  Init Containers. The example command lines below will refer to the Pod as
```
Comment (devin-donnelly) [inline docs/tasks/troubleshoot/debug-init-containers.md]: Remove "will"; avoid future tense, and also unnecessary in this case.

**Comment by devin-donnelly** [comment]
Looks great. One small nit and we can merge.

File: docs/tasks/troubleshoot/debug-init-containers.md (hunk 2 above)
Comment (enisoc) [inline docs/tasks/troubleshoot/debug-init-containers.md]: Done.

File: docs/tasks/troubleshoot/debug-init-containers.md (hunk 1 above)
Comment (smarterclayton) [inline docs/tasks/troubleshoot/debug-init-containers.md]: No such list exists, and can be open ended.

**Comment by devin-donnelly** [comment]
Still one small nit and a merge conflict (everything in StatefulSet touches the TOC, so that's okay; we just have to resolve them as we go).

**Comment by enisoc** [comment]
@devin-donnelly I resolved the merge conflict. Is there another nit, or are you talking about the one I fixed yesterday?

**Comment by devin-donnelly** [comment]
Sorry, that commit slipped past me. Should be good to go now.

**Pull request merged**